lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameLookupClient.java

```diff
@@ -122,7 +101,7 @@ public NameLookupClient(final String serverAddr,
                           final Cache<Identifier, InetSocketAddress> cache,
                           final LocalAddressProvider localAddressProvider) {
     this(serverAddr, serverPort, timeout, factory, retryCount, retryTimeout,
-        cache, localAddressProvider, new MessagingTransportFactory());
+        cache, localAddressProvider, new MessagingTransportFactory(localAddressProvider));
```

Comment:
From what I can tell, what we really want to do is deprecate all the public constructors, and leave only the package-private (no-modifier) constructor below. That way we can also remove all `MessageTransportFactory` public constructors after `0.13`. Does this make sense?